src/mtype.d

```diff
@@ -4763,7 +4764,10 @@ extern (C++) final class TypeSArray : TypeArray
                  * run on them for the size, since they may be forward referenced.
                  */
                 bool overflow = false;
-                if (mulu(tbn.size(loc), d2, overflow) >= 0x100_0000 || overflow) // put a 'reasonable' limit on it
+                
+                auto static_arr_size = mulu(tbn.size(loc), d2, overflow);
```

Comment:
@klickverbot There is strange error in compiler that prevents inlining mulu() which is annotated with `pragma(inline, true)`.
This happens *only when return value is not used* and _only in release compilations_. All compilers seem to fail, not just DMD so this seems like a strange front-end problem.
So I decided not to remove the temporary variable for the sake of making CI builds to pass.
I guess I should report it...